mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -1152,8 +1188,17 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
   using OpRewritePattern<T>::OpRewritePattern;
 
   LogicalResult matchAndRewrite(T op, PatternRewriter &b) const override {
+    bool isXdlops = false;
+    auto xdlopsV2Attr = op->template getAttrOfType<BoolAttr>("xdlopsV2");
+    if (xdlopsV2Attr && xdlopsV2Attr.getValue() == true)
+      isXdlops = true;
+    auto dataType =
+        op.input().getType().template cast<MemRefType>().getElementType();
     if (miopen::ConvOpType::Conv2DBwdDataOpType == convOpType) {
       return backwardData(op, b);
+    } else if (miopen::ConvOpType::Conv2DBwdWeightOpType == convOpType &&
+               isXdlops && dataType == b.getF32Type()) {
+      return backwardWeightAtomicAdd(op, b);
```

Comment:
comments are needed to indicate currently for backward weight we only do it in XDLOPS + f32 path.

Reply:
done